### PR TITLE
[Doc][KubeRay] Add a section about Autoscaler to the troubleshooting guide

### LINKED ISSUE #846: [Bug] Autoscaler doesn't scale to met demand
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

```
======== Autoscaler status: 2022-12-22 00:22:04.898664 ========
Node status
---------------------------------------------------------------
Healthy:
 1 head-group
 1 workergroup
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 0.0/2.0 CPU
 0.00/3.725 GiB memory
 0.00/1.046 GiB object_store_memory

Demands:
 {'CPU': 2.0}: 10000+ pending tasks/actors
 2022-12-22 00:26:46,855 WARNING resource_demand_scheduler.py:783 -- The autoscaler could not find a node type to satisfy the request: [{'CPU': 2.0}...
```

### Reproduction script

helm v0.4.0 raycluster with values.yaml:
```
image:
  tag: 2.2.0-py39-cpu

head:
  enableInTreeAutoscaling: true
  resources:
    limits:
      cpu: "1"
      memory: "2G"
    requests:
      cpu: "1"
      memory: "2G"

worker:
  minReplicas: 1
  maxReplicas: 10
  resources:
    limits:
      cpu: "1"
      memory: "2G"
    requests:
      cpu: "1"
      memory: "2G"
```

Then run this client script:
```python
from collections import Counter
import socket
import time

import ray

ray.init("ray://127.0.0.1:10001")

print('''This cluster consists of
    {} nodes in total
    {} CPU resources in total
    {} memory resources in total
'''.format(len(ray.nodes()), ray.cluster_resources()['CPU'], ray.cluster_resources()['memory']))

@ray.remote(num_cpus=2)
def f():
    time.sleep(0.001)
    # Return IP address.
    return socket.gethostbyname('localhost')

object_ids = [f.remote() for _ in range(10000)]
ip_addresses = ray.get(object_ids)

print('Tasks executed')
for ip_address, num_tasks in Counter(ip_addresses).items():
    print('    {} tasks on {}'.format(num_tasks, ip_address))
```
The script will just hang and the autoscaler will show logs as per above.

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kevin85421**
## Related issue number

Closes https://github.com/ray-project/kuberay/issues/846

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by DmitriGekhtman — APPROVED** [review]
Thank you! Nice going answering a question with a doc PR. This piece of advice should have been added to the docs years ago -- the node-pod-task hierarchy is pretty mind-bending for new users.

**Comment by DmitriGekhtman** [comment]
Oh, I guess I labelled the issue as a doc problem a while ago.
Anyways, thanks for addressing!!

**Comment by kevin85421** [comment]
> the node-pod-task hierarchy is pretty mind-bending for new users.

I agree. Perhaps we should consider making the Ray scheduler aware of K8s Pods and Nodes.

**Comment by DmitriGekhtman** [comment]
> > the node-pod-task hierarchy is pretty mind-bending for new users.
> 
> 
> 
> I agree. Perhaps we should consider making the Ray scheduler aware of K8s Pods and Nodes.

This is tricky. There has been extensive discussion in this direction but no officially endorsed solutions -- you can ask e.g. Eric L or Philipp M for details.

File: doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md [hunk 1]
```diff
@@ -66,6 +67,15 @@ RayService is a Custom Resource Definition (CRD) designed for Ray Serve. In Kube
 create Ray Serve applications once the RayCluster is ready. If the issue pertains to the data plane, specifically your Ray Serve scripts
 or Ray Serve configurations (`serveConfigV2`), troubleshooting may be challenging. See [rayservice-troubleshooting](kuberay-raysvc-troubleshoot) for more details.
 
+(autoscaler)=
+## Ray Autoscaler
+
+### Ray Autoscaler doesn't scale up, causing the new Ray tasks or actors to remain pending
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md:73]:
```suggestion
### Ray Autoscaler doesn't scale up, causing new Ray tasks or actors to remain pending
```

**Review by angelinalg — APPROVED** [review]

File: doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md (hunk 1 above)
Comment (jjyao) [inline doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md:73]: Should this be under `Questions?`

Comment (kevin85421) [reply doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md]: Updated https://github.com/ray-project/ray/pull/42962/commits/63eb27c4ded5f7878c3ba92f73585a3c61ae4fd0.

**Review by jjyao — APPROVED** [review]

**Pull request merged**